### PR TITLE
test: Add more proxy router tests

### PR DESCRIPTION
**Pull request by jachym-tousek-keboola**
Jira: https://keboola.atlassian.net/browse/PSGO-402

**Changes:**
- Added more tests

-----------

### CODE REVIEW

File: internal/pkg/service/appproxy/http/router_test.go [hunk 1]
```diff
@@ -342,30 +292,203 @@ func TestAppProxyRouter(t *testing.T) {
...
+				assert.Len(t, cookies, 1)
+				wildcards.Assert(t, "_oauth2_proxy_csrf=%s; Path=/; Domain=oidc.data-apps.keboola.local; Expires=%s; HttpOnly; Secure", cookies[0])
+
+				// Request to the OIDC provider
+				request, err := http.NewRequestWithContext(context.Background(), http.MethodGet, location, nil)
+				require.NoError(t, err)
+				response, err := client.Do(request)
+				require.NoError(t, err)
+				require.Equal(t, http.StatusFound, response.StatusCode)
+				location = response.Header["Location"][0]
```
Comment (hosekpeter) [inline internal/pkg/service/appproxy/http/router_test.go:365]: these parts of the code are repeated. Maybe you can create a function for that, but it's a detail.

**Comment by hosekpeter** [comment]
just little thing. LGTM

File: internal/pkg/service/appproxy/http/router_test.go [hunk 2]
```diff
@@ -342,30 +292,203 @@ func TestAppProxyRouter(t *testing.T) {
...
+				cookies := rec.Header()["Set-Cookie"]
+				assert.Len(t, cookies, 1)
+				wildcards.Assert(t, "_oauth2_proxy_csrf=%s; Path=/; Domain=oidc.data-apps.keboola.local; Expires=%s; HttpOnly; Secure", cookies[0])
+
+				// Request to the OIDC provider
+				request, err := http.NewRequestWithContext(context.Background(), http.MethodGet, location, nil)
+				require.NoError(t, err)
+				response, err := client.Do(request)
+				require.Error(t, err)
+				require.Nil(t, response)
```
Comment (michaljurecko) [inline internal/pkg/service/appproxy/http/router_test.go:325]: Can you please assert the error message here?
`require.Nil` is not necessary, but it is not a problem.

File: internal/pkg/service/appproxy/http/router_test.go [hunk 3]
```diff
@@ -342,30 +292,203 @@ func TestAppProxyRouter(t *testing.T) {
...
+
+				client := createNoRedirectHTTPClient()
+
+				// Request to private app (unauthorized)
+				rec := httptest.NewRecorder()
+				req := httptest.NewRequest(http.MethodGet, "https://oidc.data-apps.keboola.local/", nil)
+				handler.ServeHTTP(rec, req)
+				require.Equal(t, http.StatusFound, rec.Code)
+				location := rec.Header()["Location"][0]
+				cookies := rec.Header()["Set-Cookie"]
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/appproxy/http/router_test.go:355]: Duplicated code in tests is sometimes not a problem, if it is better for readind/editing.

**Review by michaljurecko — APPROVED** [review]
LGTM, one detail:

**Pull request merged**